### ISSUE TITLE
Notifications System results in overriding input id values

### ISSUE THREAD

**Issue by mjfaga** [mantinedev/mantine#4655]
### What package has an issue

@mantine/core

### Describe the bug

Given this input:
```
        <TextInput
          id="email"
          label="Email"
          placeholder="Enter your email"
          {...form.getInputProps('email')}
        />
```

When I render without the `@mantine/notifications` package installed, I get the following, correctly rendered input:
```
<input class="mantine-SbVPkHB mantine-Input-input mantine-TextInput-input" data-variant="default" id="email" placeholder="Enter your email" aria-invalid="false" value="">
```

However, after installing (and doing nothing else, no notifications-specific configuration or anything), the id value is overridden and the resulting input no longer respects the provided `id` value:
```
<input class="m-8fb7ebe7 mantine-Input-input mantine-TextInput-input" data-variant="default" placeholder="Enter your email" aria-invalid="false" id="mantine-3fxijoh1o-input" value="">
```

### What version of @mantine/hooks page do you have in package.json?

7.0.0-alpha.22

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/festive-shaw-6ty6yt?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by mjfaga** [comment]
It's worth calling out this issue also causing React re-hydration errors when using server-side rendering.

**Comment by rtivital** [comment]
The issue has been resolved in 7.0.0 release, it works correctly now – https://codesandbox.io/s/xenodochial-tristan-cstnq2?file=/src/App.tsx
